test/test_retry.py

```diff
@@ -0,0 +1,141 @@
+import pytest
+from mock import Mock, patch
+
+from nameko.retry import retry
+
+
+@pytest.fixture
+def tracker():
+    return Mock()
+
+
+@pytest.yield_fixture(autouse=True)
+def mock_sleep():
+    with patch('nameko.retry.sleep') as patched:
+
+        def total():
+            return sum(delay for (delay,), _ in patched.call_args_list)
+
+        patched.total = total
+        yield patched
+
+
+def test_without_arguments(tracker, mock_sleep):
+
+    @retry
+    def fn():
+        tracker()
+        raise ValueError()
+
+    with pytest.raises(ValueError):
+        fn()
+
+    assert tracker.call_count > 1
+
+
+@pytest.mark.parametrize("exceptions", [ValueError, (ValueError,)])
+def test_retry_for_exceptions(exceptions, tracker):
+    threshold = 10
+
+    @retry(for_exceptions=exceptions, max_attempts=float('inf'))
```

Comment:
No, default is 3.

I think it would be dangerous to have infinite max attempts _and_ `for_exceptions=Exception` as the defaults --
 too easy to end up retrying forever.